chall08/gbudau.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+_VALID_ACTIONS = "IKNSWE"
+_INNER_GRID = "M0*"
+_OUTER_GRID = "+E"
+_WALL = "+"
+_MINE = "*"
+_ROBOT = "M"
+_EXIT = "E"
+_INIT = "I"
+_STOP = "K"
+
+def get_minefield():
+    minefield = []
+    err = False
+
+    try:
+        line = input()
+        while line:
+            minefield.append(line)
+            line = input()
+    except EOFError:
+        err = True
+        return minefield, err
+    return minefield, err
+
+def get_actions():
+    err = False
+    actions = ""
+
+    try:
+        actions = input()
+    except EOFError:
+        err = True
+        return actions, err
+    return actions, err
+
+def equal_width(minefield):
+    length = len(minefield[0])
+
+    for row in minefield:
+        if len(row) != length: 
+            break
+    return len(row) == length
+
+def exit_in_corner(minefield):
+    height_idx = len(minefield) - 1
+    width_idx = len(minefield[0]) - 1
+
+    if minefield[0][0] == _EXIT or \
+       minefield[0][width_idx] == _EXIT or \
+       minefield[height_idx][0] == _EXIT or \
+       minefield[height_idx][width_idx] == _EXIT:
+        return True
+    return False
+
+def one_exit(minefield):
+    exits = minefield[0].count(_EXIT)
+    exits += minefield[-1].count(_EXIT)
+    
+    for row in minefield[1:-1]:
+        if exits > 1:
+            break
+        if row[0] in _EXIT or row[-1] in _EXIT:
```

Comment:
There's a problem with your check here, if the board has two exits in both sides, and it's not in the last or first row, you only count one exit :/

Which means you take:

```
+++++++++++
+*0*0*0*0*+
+0*0*0*0*0+
+000000000+
+0000M0000+
E000000000E
+++++++++++
```

as a valid map !

Reply:
Thank you, well spotted.